roofit/roofitcore/test/testRooFuncWrapper.cxx

```diff
@@ -141,6 +147,53 @@ TEST(RooFuncWrapper, NllWithObservables)
    // Check if derivatives are equal
    EXPECT_NEAR(getNumDerivative(*nllRef, mu, normSet), dMyNLL[0], 1e-6);
    EXPECT_NEAR(getNumDerivative(*nllRef, sigma, normSet), dMyNLL[1], 1e-6);
+
+   // Minimize the RooFuncWrapper Implementation
```

Comment:
This test already has some long spaghetti code, and duplicating now the code for the numerical and AD differentiations doesn't make this better.

What about introducing a helper function with a signature like this for example?

```C++
std::unique_ptr<ROOT::Math::Minimizer> doMinimization(RooFuncWrapper const&, RooArgSet const& parameters, bool useAnalyticGradient);
```

It will instantiate and setup the minimizer correctly such that the results will match with RooFits, and return the minimizer instance such that you can get `X()` and `Errors()` to do the comparisons.

You can do the calls to `SetLimitedVariable()` in a range-based loop over the `RooArgSet cons& params` argument.

As this will be for sure not be the last unit test in which we will do minimization, it would be nice if you could make this a bit more modular.